projects/ad400x-fmcz/src/ad400x_fmcz.c

```diff
@@ -78,30 +79,25 @@ int main()
 	}
 	/* Offload example */
 	else {
-		msg = (spi_eng_msg *)malloc(sizeof(*msg));
-		if (!msg)
-			return -1;
-
-		msg->spi_msg_cmds = spi_eng_msg_cmds;
-		msg->msg_cmd_len = ARRAY_SIZE(spi_eng_msg_cmds);
-		msg->rx_buf_addr = 0x800000;
-		msg->tx_buf_addr = 0xA000000;
-		msg->tx_buf[0] = 0xFF;
-		msg->tx_buf[1] = 0xFF;
+		ret = spi_engine_offload_init(dev->spi_desc, &spi_engine_offload_init_param);
+		if (ret != SUCCESS)
+			return FAILURE;
 
-		/* Init the rx buffer with 0s */
-		memset(msg->rx_buf, 0, 4 * ARRAY_SIZE(msg->rx_buf));
+		msg.commands = spi_eng_msg_cmds;
+		msg.no_commands = ARRAY_SIZE(spi_eng_msg_cmds);
+		msg.rx_addr = 0x800000;
+		msg.tx_addr = 0xA000000;
+		msg.commands_data = commands_data;
 
-		spi_eng_offload_load_msg(dev->spi_desc, msg);
-		spi_eng_transfer_multiple_msgs(dev->spi_desc, 1000);
+		ret = spi_engine_offload_transfer(dev->spi_desc, msg, AD400x_EVB_SAMPLE_NO);
+		if (ret != SUCCESS)
+			return ret;
 
 		mdelay(2000);
```

Comment:
Is there a better way to detect when the transaction is over? Like an interrupt from the DMA?